internal/service/agent/metrics_service.go

```diff
@@ -45,32 +47,60 @@ func (m *MetricsCollectionService) Send(serverAddr string) error {
 
```

Comment:
Look at the 2 lines before: you do not need to log an error if you return it from the function. Logging is a form of handling an error, and the errors are propagated to the topmost level of your code (to the place where they can be handled) and are handled there. So, in this case, you need to get the error in `main` and log it there (becase you cannot really handle it otherwise)